### PR TITLE
Performance-friendly changing of a Sprites opacity

### PR DESCRIPTION
**Pull request by dnbow**
I dont know if the fading fog is still wanted as it was a place holder so I removed it

### CODE REVIEW

**Comment by lillithrosepup** [comment]
swag

**Review by Ryos00 — APPROVED** [review]

**Review by keyraven — CHANGES_REQUESTED** [review]
This was my original method, however, it does not work properly for pixels that are already partially transparent. set_alpha() sets every pixel to the same value, which is not the intended effect.  The Starclan lineart has a lot of partial transparency.  

I also attempted to leave the lineart unchanged and only apply transparency to the color of the cat, but that effect was deemed too subtle, since it often just looks like the cat was turning black.

**Comment by dnbow** [comment]
i think this is how i got around that

File: scripts/utility.py [hunk 1]
```diff
@@ -1420,6 +1419,17 @@ def update_sprite(cat):
                 temp = sprites.sprites['fadestarclan' + stage + cat_sprite].copy()
                 temp.blit(new_sprite, (0, 0))
                 new_sprite = temp
+                
+        if cat.opacity < 100 and not cat.prevent_fading and game.settings["fading"]
```
Comment (keyraven) [inline scripts/utility.py:1423]: Missing a `:`

File: scripts/utility.py [hunk 2]
```diff
@@ -1420,6 +1419,17 @@ def update_sprite(cat):
...
+            new_sprite.set_alpha(int(cat.opacity * 2.55))
+            faded_sprite = Surface((spriteSize, spriteSize), pygame.HWSURFACE | pygame.SRCALPHA).convert_alpha()
```
Comment (keyraven) [inline scripts/utility.py]: Surface is not defined. use `pygame.Surface`

**Review by keyraven — CHANGES_REQUESTED** [review]
Concerns:

- You took the method of not applying the opacity to the lineart.  As I said in my previous review, this method was not used because it was too subtle.  The fading effect must be clear from the list screen, not just the profile screen.  It works well enough for ginger cats, but it not clear enough at the list-page level for black and gray cats.  This is not a method I want to use. 
- You are applying the lineart twice. This doubles the partial transparency for all cats with opacity applied. 
- You are applying the lineart after the masks for missing limbs are applied. This means that when cat starts fading, they gain transparent limbs.

**Comment by dnbow** [comment]
ahhhh i see, ill close it for now but if i find a solution that solves that ill reopen it

**Pull request closed without merging**